homeassistant/components/smartthings/sensor.py

```diff
@@ -212,3 +226,34 @@ def unit_of_measurement(self):
         """Return the unit this state is expressed in."""
         unit = self._device.status.attributes[self._attribute].unit
         return UNITS.get(unit, unit) if unit else self._default_unit
+
+
+class SmartThingsThreeAxisSensor(SmartThingsEntity):
+    """Define a SmartThings Three Axis Sensor."""
+
+    def __init__(self, device, index):
+        """Init the class."""
+        super().__init__(device)
+        self._index = index
+
+    @property
+    def name(self) -> str:
+        """Return the name of the binary sensor."""
+        return '{} {}'.format(
+            self._device.label, THREE_AXIS_NAMES[self._index])
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return '{}.{}'.format(
+            self._device.device_id, THREE_AXIS_NAMES[self._index])
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        from pysmartthings import Attribute
+        three_axis = self._device.status.attributes[Attribute.three_axis].value
+        try:
+            return three_axis[self._index]
+        except Exception:  # pylint:disable=broad-except
```

Comment:
We should be able to limit this exception to type error and index or key error depending on what type `three_axis` should be.

Reply:
That's right -- but do I then need to add a final `return None` for clarity outside the try/except?

Reply:
I think it's good with the returns as is.

Is `three_axis` a list or a dict? We probably only need to catch either index error or key error, besides type error, right? Type error would happen if it's `None`, I'm thinking.

Reply:
The capability schema says it _should_ be list with exactly 3 elements, but that relies on the DTH developers to follow that, which we know isn't always the case...  It appears the ST API does some normalization/coercion from dict to list, so it should be safe to drop `KeyError`.